Fix/events spark metrics provider

alfonso-vidal opened the pull request.

greeshmaswaminathan commented on cdap-app-fabric/src/main/java/io/cdap/cdap/internal/events/ProgramStatusEventPublisher.java:
```diff
@@ -144,36 +145,48 @@ protected void processMessages(StructuredTableContext structuredTableContext,
-      ProgramStatusEvent programStatusEvent = new ProgramStatusEvent(publishTime, EVENT_VERSION,
-                                                                     instanceName,
-                                                                     projectName, programStatusEventDetails);
-      programStatusEvents.add(programStatusEvent);
```
Please remove the programStatusEvents list if it is unused.

greeshmaswaminathan commented on cdap-app-fabric/src/main/java/io/cdap/cdap/internal/events/SparkProgramStatusMetricsProvider.java:
```diff
@@ -71,29 +73,29 @@ public SparkProgramStatusMetricsProvider(CConfiguration cConf) {
           if (Objects.isNull(metrics)) {
             logger.error("Error during metrics extraction");
-            return ExecutionMetrics.emptyMetrics();
+            return new ExecutionMetrics[]{ExecutionMetrics.emptyMetrics()};
```
We can just return empty array here .

greeshmaswaminathan commented on cdap-app-fabric/src/main/java/io/cdap/cdap/internal/events/SparkProgramStatusMetricsProvider.java:
```diff
@@ -71,29 +73,29 @@ public SparkProgramStatusMetricsProvider(CConfiguration cConf) {
     return Retries.supplyWithRetries(() -> {
-      ExecutionMetrics metrics;
+      ExecutionMetrics[] metrics;
       HttpResponse applicationResponse;
```
Formatting issues here

greeshmaswaminathan reviewed (approved):
Please squash the commits before merge

alfonso-vidal committed:
Add to ExecutionMetrics shuffle values and stageId
Return always an array with the different possible COMPLETE stages from spark
fix stages url
make retrieve metric asynchronous
